tremor-cli/src/server.rs

```diff
@@ -0,0 +1,280 @@
+        .post(|r| async { fix_tide(api::binding::publish_artefact(r).await) });
+    app.at("/binding/:aid")
+        .get(|r| async { fix_tide(api::binding::get_artefact(r).await) })
+        .delete(|r| async { fix_tide(api::binding::unpublish_artefact(r).await) });
+    app.at("/binding/:aid/:sid")
+        .get(|r| async { fix_tide(api::binding::get_servant(r).await) })
+        .post(|r| async { fix_tide(api::binding::link_servant(r).await) })
+        .delete(|r| async { fix_tide(api::binding::unlink_servant(r).await) });
+    app.at("/pipeline")
+        .get(|r| async { fix_tide(api::pipeline::list_artefact(r).await) })
+        .post(|r| async { fix_tide(api::pipeline::publish_artefact(r).await) });
+    app.at("/pipeline/:aid")
+        .get(|r| async { fix_tide(api::pipeline::get_artefact(r).await) })
+        .delete(|r| async { fix_tide(api::pipeline::unpublish_artefact(r).await) });
+    app.at("/onramp")
+        .get(|r| async { fix_tide(api::onramp::list_artefact(r).await) })
+        .post(|r| async { fix_tide(api::onramp::publish_artefact(r).await) });
+    app.at("/onramp/:aid")
+        .get(|r| async { fix_tide(api::onramp::get_artefact(r).await) })
+        .delete(|r| async { fix_tide(api::onramp::unpublish_artefact(r).await) });
+    app.at("/offramp")
+        .get(|r| async { fix_tide(api::offramp::list_artefact(r).await) })
+        .post(|r| async { fix_tide(api::offramp::publish_artefact(r).await) });
+    app.at("/offramp/:aid")
+        .get(|r| async { fix_tide(api::offramp::get_artefact(r).await) })
+        .delete(|r| async { fix_tide(api::offramp::unpublish_artefact(r).await) });
+
+    if !matches.is_present("no-api") {
+        eprintln!("Listening at: http://{}", host);
+        info!("Listening at: http://{}", host);
+
+        if let Err(e) = app.listen(host).await {
+            error!("API Error: {}", e);
+        }
+        warn!("API stopped");
+        world.stop().await;
+    }
+
+    handle.await;
+    warn!("World stopped");
+    Ok(())
+}
+
+fn server_run(matches: &ArgMatches) -> Result<()> {
+    version::print();
+    if let Err(ref e) = task::block_on(run_dun(matches)) {
+        error!("error: {}", e);
+        eprintln!("error: {}", e);
+        for e in e.iter().skip(1) {
+            error!("error: {}", e);
+            eprintln!("error: {}", e);
+        }
+
+        // The backtrace is not always generated. Try to run this example
+        // with `RUST_BACKTRACE=1`.
+        if let Some(backtrace) = e.backtrace() {
+            error!("backtrace: {:?}", backtrace);
+            eprintln!("backtrace: {:?}", backtrace);
+        }
+
+        // ALLOW: main.rs
+        ::std::process::exit(1);
+    } else {
+        Ok(())
+    }
+}
+
+#[cfg_attr(tarpaulin, skip)]
+pub(crate) fn run_cmd(cmd: &ArgMatches) -> Result<()> {
+    if let Some(matches) = cmd.subcommand_matches("run") {
+        server_run(matches)
+    } else {
+        // ALLOW: main.rs
```

Comment:
would be nice if just running `tremor server` gave a warning that further args are expected, like how `tremor run` works. Right now, it exits silently here.

(same for commands like `tremor api`)

Reply:
Calling `tremor server` should print help - that's a bug ... well spotted

Reply:
We can add a warning/notice to `tremor server run` though if no files are supplied for deployment on launch

Reply:
As this was referring to the `tremor server` having no help ( fixed ) marking as resolved